File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const map = new Map();
        const buckets = Array.from({length: nums.length + 1}, () => []);

        for (const num of nums) {
            map.set(num, (map.get(num) || 0) + 1);
        }

        for (const [num, freq] of map.entries()) {
            buckets[freq].push(num);
        }

        const res = [];

        for (let i = buckets.length; i > 0; --i) {
            if (!buckets[i]) {
                continue;
            }
            for (const num of buckets[i]) {
                res.push(num);
                if (res.length == k) {
                    return res;
                }
            }
        }
        return [];
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.swift
class Solution {
    func longestConsecutive(_ nums: [Int]) -> Int {
        var set = Set(nums)
        var res = 0

        for num in set {
            if !set.contains(num - 1) {
                var length = 0

                while set.contains(num + length) {
                    length += 1
                }

                res = max(res, length)
            }
        }

        return res
    }
}
